### ISSUE TITLE
Clippy craze: all variants have the same prefix/postfix

### ISSUE THREAD

**Issue by pronebird** [rust-lang/rust-clippy#16334]
### Description

Given the following very basic code:

```rs
#[derive(thiserror::Error, Debug)]
pub enum Error {
    #[error("Failed to obtain the path to cache directory.")]
    NoCacheDir,

    #[error("Failed to obtain the path to local config directory.")]
    NoLocalConfigDir,

    #[error("Failed to obtain the path to download directory.")]
    NoDownloadDir,
}
```

First I stumble upon:

> warning: all variants have the same prefix: `No`

When I remove "No" just because the damn thing wouldn't shut up, I get back to where I started:

> warning: all variants have the same postfix: `Dir`

This clippy rule forced me to rewrite my code to:

```rs
#[derive(Debug)]
enum DirKind {
    Cache,
    LocalConfig,
    Download,
}

impl std::fmt::Display for DirKind {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_str(match self {
            Self::Cache => "cache",
            Self::LocalConfig => "local config",
            Self::Download => "download",
        })
    }
}

#[derive(thiserror::Error, Debug)]
#[error("Failed to obtain the path to {kind} directory.")]
pub struct Error {
    kind: DirKind,
}

impl Error {
    fn new(kind: DirKind) -> Self {
        Error { kind }
    }
}
```

However arguably there is very little gain in doing so. This is really a waste of time in my opinion. Maybe these rules should be made optional. What do you think?

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
```

**Comment by blyxyas** [comment]
The lint that triggered these warnings is [`enum_variant_names`](https://rust-lang.github.io/rust-clippy/rust-1.92.0/index.html#enum_variant_names).

Did the lint originate from the `Derive(thiserror::Error)` proc-macro?

What about:

```rust
#[derive(thiserror::Error, Debug)]
pub enum PathNotFoundError {
    #[error("Failed to obtain the path to cache directory.")]
    Cache,

    #[error("Failed to obtain the path to local config directory.")]
    LocalConfig,

    #[error("Failed to obtain the path to download directory.")]
    Download,
}
```

Still, if you'd like, you can always use `#[allow(clippy::enum_variant_names)]` or even `#![allow(clippy::enum_variant_names)]` to allow it globally if you use this pattern a lot.

**Comment by CommanderStorm** [comment]
I am not entirely sure if there is something to be done here.

The lint also has two differnt setings, which should be default to a way which likely insentivises better code..

- `avoid-breaking-exported-api`: Suppress lints whenever the suggested change would cause breakage for other crates. (default: true)
- `enum-variant-name-threshold`: The minimum number of enum variants for the lints about variant names to trigger (default: 3)

**Comment by blyxyas** [comment]
I'll close this because I don't think we can do anything about it and it has gone stale. I hope that the issue has been resolved on your end, if not please look into either:

- Adding an unknown, unreachable or otherwise variant to your enum,
- Allowing the lint for the enum.

Thank you for the issue report! (=ↀωↀ=) ❤️
